app/src/main/java/co/infinum/rxpokemon/ui/login/LoginPresenter.java

```diff
@@ -51,35 +51,33 @@ public void login(String username, String password) {
 
         final LoginParams params = new LoginParams(username, password);
 
-        cancel();
+        interactor.execute(params)
+                .subscribeOn(ioScheduler)
+                .observeOn(mainThreadScheduler)
+                .retry(NUMBER_OF_RETRIES)
+                .subscribeWith(new RxDisposableObserver<LoginResponse>(errorHandler, compositeDisposable) {
 
-        loginDisposable =
-                interactor.execute(params)
-                        .subscribeOn(ioScheduler)
-                        .observeOn(mainThreadScheduler)
-                        .retry(NUMBER_OF_RETRIES)
-                        .subscribeWith(new RxDisposableObserver<LoginResponse>(errorHandler) {
+                    @Override
+                    public void onNext(LoginResponse loginResponse) {
+                        User.getInstance().setEmail(loginResponse.getEmail());
+                        User.getInstance().setAuthToken(loginResponse.getAuthToken());
+                        view.setState(new LoginViewState(State.LOGIN_COMPLETED));
+                    }
 
-                            @Override
-                            public void onNext(LoginResponse loginResponse) {
-                                User.getInstance().setEmail(loginResponse.getEmail());
-                                User.getInstance().setAuthToken(loginResponse.getAuthToken());
-                                view.setState(new LoginViewState(State.LOGIN_COMPLETED));
-                            }
 
-                            @Override
-                            public void onComplete() {
+                    @Override
+                    public void onComplete() {
+
+                    }
+                });
 
-                            }
-                        });
 
     }
 
     @Override
     public void cancel() {
-
-        if (loginDisposable != null && !loginDisposable.isDisposed()) {
-            loginDisposable.dispose();
+        if (compositeDisposable != null && !compositeDisposable.isDisposed()) {
```

Comment:
@kjurkovic @reisub Mislim da bi umjesto dispose ovdje trebali koristiti clear. Clear će maknuti sve reference na disposable objekte koje ovaj container ima i disposati te objekte. Ako se direktno disposea ovaj objekt onda se više ne može reusati, što znači ako app ode u background mora se napraviti nova instance jer inače neće raditi.